src/Sarif.Multitool/Rules/SARIF2009.UseConventionalSymbolicNames.cs

```diff
@@ -0,0 +1,105 @@
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF2009_UseConventionalSymbolicNames_FullDescription_Text };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.SARIF2009_UseConventionalSymbolicNames_Warning_UseConventionalRuleIds_Text),
+            nameof(RuleResources.SARIF2009_UseConventionalSymbolicNames_Warning_UseConventionalUriBaseIdNames_Text)
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+        private static readonly string[] s_conventionalSymbols = new string[] { "REPOROOT", "SRCROOT", "TESTROOT", "BINROOT" };
+        private static readonly Regex s_conventionalIdRegex = new Regex(@"^[A-Z]{1,5}[0-9]{1,4}$", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        protected override void Analyze(Run run, string runPointer)
+        {
+            if (run.OriginalUriBaseIds != null)
+            {
+                AnalyzeOriginalUriBaseIds(run.OriginalUriBaseIds, runPointer.AtProperty(SarifPropertyName.OriginalUriBaseIds));
+            }
+        }
+
+        protected override void Analyze(Tool tool, string toolPointer)
+        {
+            if (tool.Driver != null)
+            {
+                AnalyzeToolDriver(tool.Driver, toolPointer.AtProperty(SarifPropertyName.Driver));
+            }
+        }
+
+        private void AnalyzeToolDriver(ToolComponent toolComponent, string toolDriverPointer)
+        {
+            if (toolComponent.Rules != null)
+            {
+                string rulesPointer = toolDriverPointer.AtProperty(SarifPropertyName.Rules);
+                for (int i = 0; i < toolComponent.Rules.Count; i++)
+                {
+                    AnalyzeReportingDescriptor(toolComponent.Rules[i], rulesPointer.AtIndex(i));
+                }
+            }
+        }
+
+        private void AnalyzeReportingDescriptor(ReportingDescriptor reportingDescriptor, string reportingDescriptorPointer)
+        {
+            if (string.IsNullOrWhiteSpace(reportingDescriptor.Id))
+            {
+                return;
+            }
+
+            if (!s_conventionalIdRegex.IsMatch(reportingDescriptor.Id))
+            {
+                // {0}: The 'id' property of the rule '{1}' does not follow the recommended format:
+                // a short string identifying the tool concatenated with a numeric rule number, for example, `CS2001`.
+                // Using a conventional format for the rule id provides a more uniform experience across tools.
+                LogResult(
+                    reportingDescriptorPointer.AtProperty(SarifPropertyName.Id),
+                    nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_IncludeDynamicContent_Text),
```

Comment:
>SARIF2001_AuthorHighQualityMessages_Warning_IncludeDynamicContent_Text [](start = 41, length = 70)

resource name wrong?

Reply:
just updated! thanks!

---
In reply to: [445549394](https://github.com/microsoft/sarif-sdk/pull/1938#discussion_r445549394) [](ancestors = 445549394)